### PR TITLE
Move `experiment_asset.json` and `file_asset.json` schema into new structure

### LINKED ISSUE #4: Versioning schemas
As we may be adding schemas for checksums and performance metrics to this repository, it would be good to have standard way to handle different schemas versions. Some options are:

1. **Schemas per branch and specific schema git tags.** So a repository organisation similar to Cosima's [ACCESS-OM3 configurations](https://github.com/COSIMA/MOM6-CICE6). For example, for the existing schema `experiment_asset.json`, there's a branch name called `experiment_asset` and related tag versions of `experiment_asset-v1.0.0`. Descriptions of schema would need to be in the README of the `main` branch for discoverability. This will have the benefit of separate git histories. 
2. **Just specific schema git tags.** For example, a tag could be `experiment_asset-v1.0.0` but have all the files in the `main` branch. +: Schema visibility, git history for 1 file. -: Shared git history, long tags.
3. **Creating separate files per schema version.**. For example, `experiment-asset-v1.0.json` or directory structure of:
```
─ experiment-asset
│     v1.0.json
│     v1.1.json
─ schema2 
```
+: Schema and version visibility. -: Need to guarantee that earlier versions of schema do not change,  no git history of 1 file.
4. **Creating a separate repositories per schema.** +: Simple tags and separate git histories. -: no central location (would git submodules work here?)

Thanks to @dougiesquire and @aidanheerdegen for the ideas! I have a slight preference for option 1 but I think any option would work fine.

### PR DESCRIPTION
**Pull request by dougiesquire**
- Renamed to `experiment-metadata` and `file-metadata`
- `schema_version` has been added to the schema properties but are not included in `required`. This is because doing so would mean that all `metadata.yaml` files currently being used by the ACCESS-NRI Intake catalogue would need to be modified.
- I've left a copy of the original `experiment_asset.json` in the base directory for the time being. This is because removing this will [break Payu](https://github.com/payu-org/payu/blob/421431b52c24dd57fa0ad023d46b739667133af7/payu/metadata.py#L39). Will delete once a new Payu release (or a few if preferred) that works with the new structure has been made. 

(re) Closes #4

### CODE REVIEW

**Comment by aidanheerdegen** [comment]
> * `schema_version` has been added to the schema properties but are not included in `required`. This is because doing so would mean that all `metadata.yaml` files currently being used by the ACCESS-NRI Intake catalogue would need to be modified.

Does that mean there should be a 2-0-0 version with `schema_version` set to `required?

> * I've left a copy of the original `experiment_asset.json` in the base directory for the time being. This is because removing this will [break Payu](https://github.com/payu-org/payu/blob/421431b52c24dd57fa0ad023d46b739667133af7/payu/metadata.py#L39). Will delete once a new Payu release (or a few if preferred) that works with the new structure has been made.

I've made a [related payu issue](https://github.com/payu-org/payu/issues/429) but TL;DR if we'd referenced a specific commit of the schema it wouldn't matter if it was a deleted from the repo, e.g.

https://raw.githubusercontent.com/ACCESS-NRI/schema/36abe2fe28eb2853a54f41c5eedfd964617d9d68/experiment_asset.json

This should be the standard approach I think. It is a bit cumbersome, but it makes the schema less vulnerable to changes.

**Comment by dougiesquire** [comment]
> Does that mean there should be a 2-0-0 version with schema_version set to `required?

I guess... But adding non-interoperable changes to a schema that validates potentially hundreds of `metadata.yaml` files is going to be a _massive_ pain in the bum. So I'd be unlikely to update to using 2-0-0 for the ACCESS-NRI Intake catalog anytime soon... Not really sure what the best way forward here is.

> This should be the standard approach I think. It is a bit cumbersome, but it makes the schema less vulnerable to changes.

Yeah, as I mentioned [here](https://github.com/ACCESS-NRI/schema/issues/4#issuecomment-1998998276) this is what I did for the ACCESS-NRI Intake catalog

**Comment by aidanheerdegen** [comment]
> But adding non-interoperable changes to a schema that validates potentially hundreds of `metadata.yaml` files is going to be a _massive_ pain in the bum. So I'd be unlikely to update to using 2-0-0 for the ACCESS-NRI Intake catalog anytime soon... Not really sure what the best way forward here is.

I don't think there is a downside to making a `2-0-0` version and start using that for new `metadata.yaml` files. They validate against the schema listed internally right?

> Yeah, as I mentioned [here](https://github.com/ACCESS-NRI/schema/issues/4#issuecomment-1998998276) this is what I did for the ACCESS-NRI Intake catalog

Indeed. Apologies for rediscovering what you'd already pointed out. This should be documented somewhere. Could you maybe add that to the README?

File: au.org.access-nri/model/output/experiment-metadata/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Experiment metadata
+
+This schema is used for metadata associated with a model experiment. A model experiment is loosely defined as a collection of one or more climate model datasets that each may comprise one a more files. Examples of a model experiment include:
```
Comment (aidanheerdegen) [inline au.org.access-nri/model/output/experiment-metadata/README.md:3]:
```suggestion
This schema is used for metadata associated with a model experiment. A model experiment is loosely defined as a collection of one or more climate model datasets that each may comprise one or more files. Examples of a model experiment include:
```

**Review by aidanheerdegen — APPROVED** [review]
LGTM

**Comment by dougiesquire** [comment]
> I don't think there is a downside to making a 2-0-0 version and start using that for new metadata.yaml files. They validate against the schema listed internally right?

Okey doke. So just to check I understand properly:

- Add `2-0-0.json` versions with `schema_version` field required
- Use 1-0-0 versions in `access-nri-intake`
- Use 2-0-0 version in Payu (and have Payu write the `schema_version` field)

Note that the `metadata-template` and `metadata-validate` [cmdline scripts recently exposed in `access-nri-intake`](https://github.com/ACCESS-NRI/access-nri-intake-catalog/pull/158) will use 1-0-0.

> Could you maybe add that to the README?

Good idea

**Comment by aidanheerdegen** [comment]
> Okey doke. So just to check I understand properly:
> 
> * Add `2-0-0.json` versions with `schema_version` field required
> * Use 1-0-0 versions in `access-nri-intake`
> * Use 2-0-0 version in Payu (and have Payu write the `schema_version` field)

Yes. I think.

> Note that the `metadata-template` and `metadata-validate` [cmdline scripts recently exposed in `access-nri-intake`](https://github.com/ACCESS-NRI/access-nri-intake-catalog/pull/158) will use 1-0-0.

You see this is where my understanding hits a knowledge cliff. I thought they would validate against the schema they use internally, so you could incrementally update the `metadata.yaml` to version 2-0-0.

**Comment by dougiesquire** [comment]
> You see this is where my understanding hits a knowledge cliff. I thought they would validate against the schema they use internally, so you could incrementally update the metadata.yaml to version 2-0-0.

That's not what currently happens. The tools were written for the `access-nri-intake`, so they validate/template whatever version is being used by `access-nri-intake` (i.e. 1-0-0 in the case proposed [above](https://github.com/ACCESS-NRI/schema/pull/12#issuecomment-2002793207)).

**Comment by aidanheerdegen** [comment]
> That's not what currently happens. The tools were written for the `access-nri-intake` package, so they validate/template whatever version is being used by `access-nri-intake` (i.e. 1-0-0 in the case proposed [above](https://github.com/ACCESS-NRI/schema/pull/12#issuecomment-2002793207)).

I think that is just me getting myself horribly confused. So .. will it be a problem to validate `2-0-0` against a `1-0-0` schema? Is `2-0-0` a super-set of `1-0-0`. Is there any point to this point?

**Review by aidanheerdegen — DISMISSED** [review]
LGTM

**Comment by dougiesquire** [comment]
> So .. will it be a problem to validate 2-0-0 against a 1-0-0 schema

No, that won't be a problem. My point is a minor one:
- We'll want to use 1-0-0 in `access-nri-intake` since `access-nri-intake` uses the schema to validate the `metadata.yaml` file for each of the experiments in the catalog. Updating to using 2-0-0 in `access-nri-intake` will break the catalog build since existing experiment `metadata.yaml`s won't validate against 2-0-0 (because they're missing `schema_version`). 
- This means that the [cmdline tools recently exposed in `access-nri-intake`](https://access-nri-intake-catalog.readthedocs.io/en/latest/management/building.html#metadata-yaml-files) will use 1-0-0. So calling `metadata-validate` will not ensure that `schema_version` is present.

**Comment by dougiesquire** [comment]
Also, apologies - I'm about to make a couple of changes in response to your first review:

- Add a `2-0-0.json`
- Add something to README about using specific commits of schema

so Github may require another review.

**Comment by dougiesquire** [comment]
Ready (again)

**Review by aidanheerdegen — APPROVED** [review]
Sill LGTM

**Pull request merged**